src/main/java/com/mineboundteam/minebound/magic/OffensiveSpells/NecroticOffensiveSpell.java

```diff
@@ -0,0 +1,159 @@
+import com.mineboundteam.minebound.magic.MagicType;
+import com.mineboundteam.minebound.magic.SpellType;
+
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.food.FoodData;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.TooltipFlag;
+import net.minecraft.world.level.Level;
+import net.minecraftforge.common.ForgeConfigSpec;
+import net.minecraftforge.common.ForgeConfigSpec.DoubleValue;
+import net.minecraftforge.common.ForgeConfigSpec.IntValue;
+import net.minecraftforge.event.entity.living.LivingDamageEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.config.ModConfigEvent;
+
+
+@Mod.EventBusSubscriber(modid = MineBound.MOD_ID)
+public class NecroticOffensiveSpell extends ActiveSpellItem {
+    public static DoubleValue FOOD_REDUCTION;
+
+    private final int manaCost;
+    private final double damageBoost;
+    private final double vampirePercent;
+
+    public final static String ACTIVE_TAG = MineBound.MOD_ID + ".necrotic_offensive.active";
+    public final static String TICK_TAG = MineBound.MOD_ID + ".necrotic_offensive.ticks";
+
+    public NecroticOffensiveSpell(Properties properties, NecroticOffensiveSpellConfig config) {
+        super(properties, config.LEVEL, MagicType.NECROTIC, SpellType.OFFENSIVE);
+        manaCost = config.MANA_COST.get();
+        damageBoost = config.DMG_BOOST.get();
+        vampirePercent = config.HEALTH_AMT.get();
+    }
+
+    @Override
+    public void use(ItemStack stack, Level level, Player player) {
+        stack.getOrCreateTag().putBoolean(ACTIVE_TAG, true);
+    }
+
+    @Override
+    public void onUsingTick(ItemStack stack, Level level, Player player) {
+        if(stack.getOrCreateTag().getBoolean(ACTIVE_TAG)) {
+            int ticks = stack.getOrCreateTag().getInt(TICK_TAG);
+            if(ticks < 20)
+                ticks++;
+            else {
+                reduceMana(manaCost, player);
+                ticks = 0; 
+            }
+            stack.getOrCreateTag().putInt(TICK_TAG, ticks);
+        }
+    }
+
+    @Override
+    public void releaseUsing(ItemStack stack, Level level, Player player) {
+        stack.getOrCreateTag().putBoolean(ACTIVE_TAG, false);
+        stack.getOrCreateTag().putInt(TICK_TAG, 0);
+    }
+
+    @SubscribeEvent
+    public static void onAttack(LivingDamageEvent event){
+        if(event.getSource().getDirectEntity() != null && event.getSource().getDirectEntity() instanceof ServerPlayer player){
+            ItemStack spell = ItemStack.EMPTY; 
+            if(player.getUseItem().getItem() instanceof NecroticOffensiveSpell)
+                spell = player.getUseItem();
+            else {
```

Comment:
Technically, we shouldn't have to check this since you can't use an item and attack at the same time. But I guess doesn't hurt just to be complete.